tests/test_fragments.py

```diff
@@ -378,3 +378,20 @@ def test_detect_blank_line(writing_string, expected_status):
     """Create some strings and then see blank detection for the line works"""
     found_blanks = fragments.is_blank_line(writing_string)
     assert found_blanks == expected_status
+
+
+@pytest.mark.parametrize(
+    "regex_fragment,expected_status", [("test", True), ("[", False)]
+)
+def test_is_valid_regex(regex_fragment, expected_status):
+    """Checks that regex assessment correctly validates regular expressions"""
+    regex_status = fragments.is_valid_regex(regex_fragment)
+    assert regex_status == expected_status
+
+
+@pytest.mark.parametrize("regex_fragment,expected_status", [("[", True)])
```

Comment:
Many more test cases are needed here!
This is not even verifying that the count is correct!

You should look to the fragment tests as examples.

Reply:
@leonardoz15 is working to finish these soon and will resolve the conversation once he has done so.